circle.yml

```diff
@@ -17,7 +17,13 @@ test:
     - nvm install 5 && npm test
 
 deployment:
+  development:
+    branch: /^(?!master).*$/
```

Comment:
`*` - 0 or more of the preceding character
`.` - any character except line breaks

So without the `.` it will try to match 0 or more of _nothing_, which matches nothing.
You can play with it here and some example branch names: http://regexr.com/3d1vg